kernel/src/sched.rs

```diff
@@ -900,4 +732,236 @@ impl Kernel {
 
         (return_reason, time_executed_us)
     }
+
+    #[inline]
+    unsafe fn handle_syscall_fired<P: Platform>(
```

Comment:
I would also like to investigate marking this function `unsafe`. It seems like the unsafe parts might be better encapsulated in a specific block.